wemake_python_styleguide/violations/consistency.py

```diff
@@ -2343,3 +2345,37 @@ class AssignToSliceViolation(ASTViolation):
 
     error_template = 'Found assignment to a subscript slice'
     code = 362
+
+
+@final
+class UnsafeGeneratorExpressionViolation(ASTViolation):
+    """
+    Forbids the reassignment of variables that affect the generator expression.
+
+    Reasoning:
+        Reassigning values to variables that affect a generator expression
+        before the generator expression is used will produce a different
+        result from what it is expected.
+
+    Solution:
+        Either do not reassign new values to the variables that affect the
+        generator expression or implement it before reassigning.
+
+    Example:
+        # Correct
```

Comment:
Can you please use more meaningful names here? We try to write our correct examples as close to the real-world code as possible.